src/components/PopupWithForm.js

```diff
@@ -20,14 +30,29 @@ export default class PopupWithForm extends Popup {
     return fromValues;
   }
 
-  setEventListners() {
-    super.setEventListners();
-    this._popupForm.addEventListener("submit", (e) => {
-      this._handleFormSubmit(this._getinputvalues());
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit",  (e) => {
+      e.preventDefault();
+      this.setLoading(true);
+      this._submitButton.disabled = true;
+      const formData = this._getinputvalues();
+      //setTimeout(() => {
+       this._handleFormSubmit(formData);
+      //  this._cardFormSubmit(formData);
+      //  this._submitButton.textContent = this._defaultButtonText;
+      //  this._submitButton.disabled = false;
+      //} 3000);
       this._popupForm.reset();
```

Comment:
Please, delete `this._popupForm.reset();` from line 46 in `PopupWithForm`


You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.